build.py

```diff
@@ -1102,6 +1102,16 @@ def dockerfile_prepare_container_linux(argmap, backends, enable_gpu,
     if 'onnxruntime' in backends:
         df += '''
 ENV LD_LIBRARY_PATH /opt/tritonserver/backends/onnxruntime:${LD_LIBRARY_PATH}
+'''
+
+    # Necessary for libtorch.so to find correct HPCX libraries
+    if ('pytorch' in backends):
+        if not enable_gpu:
+            df += '''
+ENV LD_LIBRARY_PATH /opt/hpcx/ucc/lib/:${LD_LIBRARY_PATH}
+'''
```

Comment:
Can we remove this part? Otherwise looks good.

Reply:
We cannot, or the CPU-only image will not be able to find libucc.so.1 (in the /opt/hpcx/ucc/lib folder). With the other version of these changes, it wasn't necessary since the library was added to the backend directory, which was on the RPATH. Since this is now copying over to the /opt/hpcx/ folder, the CPU-only build needs to know where to find library or it will have the previous failure to load.

If easier to read, I can combine this with the other LD_LIBRARY_PATH. I broke it up like this though, because I didn't want to modify the LD_LIBRARY_PATH in GPU images since we don't have to.

Reply:
Why do we only include ucc but not ucx. in cpu-only build. Is that a typo?

Reply:
That's why I'm a bit confused.

Reply:
We include both. There's no else statement. For the CPU-only build, we need to both the ucc and ucx on the path. For the GPU-only build, we only need the ucx path (since the ucc is found automatically).